docs/content/ingestion/tasks.md

```diff
@@ -245,10 +234,55 @@ The grammar is:
     "dataSource": <task_datasource>,
     "aggregations": <list of aggregators>,
     "rollup": <whether or not to rollup data during a merge>,
-    "interval": <DataSegment objects in this interval are going to be merged>
+    "interval": <DataSegment objects in this interval are going to be merged>,
+    "context": <task context>
 }
 ```
 
+### Compaction Task
+
+Compaction tasks merge all segments of the given interval. The syntax is:
+
+```json
+{
+    "type": "compact",
+    "id": <task_id>,
+    "dataSource": <task_datasource>,
+    "interval": <interval to specify segments to be merged>,
+    "tuningConfig" <index task tuningConfig>,
+    "context": <task context>
+}
+```
+
+|Field|Description|Required|
+|-----|-----------|--------|
+|`type`|Task type. Should be `compact`|Yes|
+|`id`|Task id|No|
+|`dataSource`|dataSource name to be compacted|Yes|
+|`interval`|interval of segments to be compacted|Yes|
+|`tuningConfig`|[Index task tuningConfig](#tuningconfig)|No|
+|`context`|[Task context](#taskcontext)|No|
+
+An example of compaction task is
+
+```json
+{
+  "type" : "compact",
+  "dataSource" : "wikipedia",
+  "interval" : "2017-01-01/2018-01-01"
+}
+```
+
+This compaction task merges _all segments_ of the interval `2017-01-01/2018-01-01` into a _single segment_.
+To merge each day's worth of data into a separate segment, you can submit multiple `compact` tasks, one for each day. They will run in parallel.
+
+A compaction task internally generates an `index` task spec for performing compaction work with some fixed parameters.
+For example, its `firehose` is always the [ingestSegmentSpec](./firehose.html) and `dimensionsSpec` and `metricsSpec`
+always include all dimensions and metrics of the input segments.
+
+Note that the output segment is rolled up only when `rollup` is set for all input segments.
+See [Segment Metadata Queries](../querying/segmentmetadataquery.html#analysistypes) for more details about `rollup`.
```

Comment:
Can you add a reference to http://druid.io/docs/latest/design/index.html#roll-up here, and add a comment to the link to the SegmentMetadataQuery docs about how that query can be used to determine whether a segment was created with rollup or not? 

The "design" link has a more substantial explanation of what rollup is, and the extra comment re: SegmentMetadataQuery would make it more clear what it's used for